lib/esLoader.js

```diff
@@ -7,7 +7,8 @@ var Writable = require('readable-stream/writable');
 function writeStream(client, index, type){
   Writable.call(this);
   this.client = client;
-  this.index = index;
+  this.canonicalIndex = index;
+  this.internalIndex = Math.round(Math.random()*1e15).toString() + Math.round(Math.random()*1e15).toString();
```

Comment:
It seems like it would be useful to have a human readable `internalIndex` name that at least indicates the `index` name and a sequential value comparable to other `internalIndex`es.  Something like `{{index_name}}-{{timestamp}}`?  Maybe...

```javascript
this.internalIndex = index + '-' + (new Date()).toISOString())
```

...resulting in...

    state-2015-05-07T22:38:52.394Z

Not sure if that's even a valid index name in ES...